Formula/g2o.rb

```diff
@@ -0,0 +1,60 @@
+class G2o < Formula
+  desc "General framework for graph optimization"
+  homepage "https://openslam-org.github.io/g2o.html"
+  url "https://github.com/RainerKuemmerle/g2o/archive/refs/tags/20201223_git.tar.gz"
+  version "20201223"
+  sha256 "20af80edf8fd237e29bd21859b8fc734e615680e8838824e8b3f120c5f4c1672"
+  license "BSD-2-Clause"
+
+  depends_on "cmake" => :build
+  depends_on "eigen"
+
+  resource "testdata" do
+    url "https://raw.githubusercontent.com/OpenSLAM-org/openslam_g2o/2362b9e1e9dab318625cd0af9ba314c47ba8de48/data/2d/intel/intel.g2o"
+    sha256 "4d87aaf96e1e04e47c723c371386b15358c71e98c05dad16b786d585f9fd70ff"
+  end
+
+  def install
+    cmake_args = std_cmake_args + %w[-DG2O_BUILD_EXAMPLES=OFF]
+    # For Intel: manually set desired SSE features to enable support for older machines.
+    # See https://gcc.gnu.org/onlinedocs/gcc/x86-Options.html for supported CPU features
+    if Hardware::CPU.intel?
+      cmake_args << "-DDO_SSE_AUTODETECT=OFF"
+      cmake_args += if OS.mac?
+        # On macOS, the earliest supported architecture is nehalem. assert so that we get
+        # a reminder to re-visit if this changes.
+        assert_equal Hardware.oldest_cpu, :nehalem
+        %w[-DDISABLE_SSE4_A=ON]
+      else
+        # Otherwise, the earliest supported architecture is core2
+        assert_equal Hardware.oldest_cpu, :core2
+        %w[-DDISABLE_SSE4_1=ON -DDISABLE_SSE4_2=ON -DDISABLE_SSE4_A=ON]
+      end
```

Comment:
How about a
```ruby
case Hardware.oldest_cpu
```
? That way you can just error out in `else`. I'm not actually sure `assert_equal` works in `install`?

Reply:
Took me a bit of a think to figure out what you meant by this suggestion. Sure, makes sense 😄